Src/Newtonsoft.Json.Schema/Infrastructure/Validation/ObjectScope.cs

```diff
@@ -16,9 +16,16 @@ internal sealed class ObjectScope : SchemaScope
     {
         private int _propertyCount;
         private string? _currentPropertyName;
-        private List<string>? _requiredProperties;
+        private JSchemaDictionary? _insensitivePropertySchemas;
+        private JSchemaDependencyDictionary? _insensitiveDependencySchemas;
+        private Dictionary<string, IList<string>>? _insensitiveDependentRequired;
+        private ICollection<string>? _requiredProperties;
         private Dictionary<string, UnevaluatedContext>? _unevaluatedScopes;
-        internal List<string>? ReadProperties;
+        internal ICollection<string>? ReadProperties;
+
+        private JSchemaDictionary? PropertySchemas => Context.Validator.PropertyNameCaseInsensitive ? _insensitivePropertySchemas : Schema._properties;
+        private JSchemaDependencyDictionary? DependencySchemas => Context.Validator.PropertyNameCaseInsensitive ? _insensitiveDependencySchemas : Schema._dependencies;
+        private Dictionary<string, IList<string>>? DependentRequired => Context.Validator.PropertyNameCaseInsensitive ? _insensitiveDependentRequired : Schema._dependentRequired;
```

Comment:
what's the difference between Context.Validator here and the one below in the Initialize method? Is the initialize called on a different cadence than the static Context.Validator updates?

Reply:
They are actually the same. Initialize()@L32 sets `Context` in the base (it's not static) to the context parameter passed.